core/dbt/materializations/incremental/microbatch.py

```diff
@@ -18,6 +18,7 @@ def __init__(
         is_incremental: bool,
         event_time_start: Optional[datetime],
         event_time_end: Optional[datetime],
+        batch_current_time: Optional[datetime] = None,
     ):
         if model.config.incremental_strategy != "microbatch":
```

Comment:
Are we in the right branch for `batch_current_time`?

Reply:
I'm not sure I understand, can you elaborate on "in the right branch"?

Reply:
If this is in reference to `batch` vs `microbatch`. For better or worse, we've been kinda using them interchangeably 🙈

Reply:
### TL;DR
`batch_current_time` (now renamed to `default_end_time`) is _only_ relevant to microbatch models

---

This file (`microbatch.py`) and the class itself (`MicrobatchBuilder`) should only ever be invoked when running on a microbatch model. That logic for separating whether we in a "default" (a.k.a. non-microbatch) model or a microbatch model lives [in run.py in the execute() method](https://github.com/dbt-labs/dbt-core/blob/fc8eb820aa97aca6da8bafe603f4007167686eb6/core/dbt/task/run.py#L459). Thus, if we're in this code (microbatch.py), we're inherently in a microbatch context / logic branch. Hope that all makes sense. I sometimes forget that the rest of the team doesn't automatically have the context of how microbatch models work 😅 Let me know if you want a walk through the microbatch code at large if that'd be useful 🙂